.github/workflows/ci.yml

```
@@ -22,10 +22,21 @@ jobs:
       uses: actions/checkout@v2
       with:
         submodules: true
-    - name: Build ${{ matrix.distro }} Docker image with ${{ matrix.compiler }}
+    - name: Prepare base image in Docker
+      run: docker build . -f utils/docker/Dockerfile.${{ matrix.distro }}
+                          --build-arg DAOS_DEPS_BUILD=no
```

Use the more natural boolean logic available in the Bourne shell:
```suggestion
                          --build-arg DAOS_DEPS_BUILD=false
```

---

This is also driven from groovy in pipeline lib (the dockerfiles, not the github yaml obviously, so it's not all that obvious is using groovy native types would translate correctly to bash native types.

---

But are you using groovy native (boolean) types in pipeline-lib or `yes`/`no` string comparisons?  If string comparisons, you can use `true`/`false` as strings equally well as `yes`/`no`.

Are you referring to https://github.com/daos-stack/pipeline-lib/blob/60a5e839a8144dbb38c7d558d915e219e6bbdc4d/vars/dockerBuildArgs.groovy#L76-L84 which looks like it's just building strings, or is there some boolean logic somewhere I'm not recalling?

---

I think it's all strings.  I was worried that if I used native groovy types then it wouldn't translate directly into bash types, although I didn't test that.

---

Right.  So I am not at all suggesting using native groovy types (at least until we understand their boolean type a bit better) but rather using posix shell native boolean logic (`true`/`false`).  That is super easy to do since we are not at this point trying to use native groovy types and are just representing those values as groovy string types currently `yes` and `no`.  But being strings, those could just as easily be `true`/`false` which makes the shell more natural to read and eliminates the execution of the `[` command and it's string match test.

---

This might have been preferable however this is also a value that we're accepting free-form from the user on the command line, so accepting a wider set of values (or failing better to be more precise) is no bad thing, and in Jenkins the values are provided by pipeline-lib so I think we're stuck with "yes" vs != "yes" for now.

---

If it's being accepted free-form then we either (a) have to validate it or (b) assume they know the right value to use.  Since we are not doing (a), expecting the user to know the values are `true`/`false` is no different than the `yes`/`no`.  Some percentage of users are going to guess right and some wrong.  The latter will have to figure out that they used the wrong boolean words.

As for pipeline-lib, it's a pity that `yes`/`no` were used.  But that's [not unfixable](https://github.com/daos-stack/pipeline-lib/pull/196) particularly given the inconsistent use of `true`/`false`/`yes`/`no` in that function currently.

---

I just mean that developers or occasionally users will be passing it on the docker command line.  Previously we had NOBUILD that you just had to define to anything to disable the build, now we have DAOS_DEPS_BUILD, DAOS_BUILD and DAOS_JAVA_BUILD that all default to "yes" and have to be set to anything other than that to disable that part of the build.

I see QUICKBUILD is already using true/false.

Using "docker build ... --build-arg QUICKBUILD=no" returns "/bin/sh: no: command not found" and then does not stop the image building so I'm not convinced that using native types is better.

---

Yes, those are exactly my points.  There is no validation happening for any arg, so it's a crapshoot whether the user gets it right or not no matter which we use.

But what makes it even worse now is that, as you point out, QUICKBUILD has already established the use of `true` and `false` for booleans and now more have been added that are `yes` and `no`.

Anyway, I'm not going to beat this dead horse any more.